src/components/Calculator.js

```diff
@@ -1,54 +1,100 @@
-// import './Calculator.css';
 import React from 'react';
+import calculate from '../logic/calculate';
+
+export default class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((state) => calculate(state, e.target.textContent));
+  }
 
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
   render() {
+    const { next, total } = this.state;
     return (
       <div className="calculator">
         <table className="table">
           <thead>
             <tr>
-              <td colSpan="4" className="result">0</td>
+              <td colSpan="4" className="result">{next || total || 0}</td>
             </tr>
           </thead>
           <tbody>
             <tr>
-              <td className="number">AC</td>
-              <td className="number">±</td>
-              <td className="number">%</td>
-              <td className="calcSign">÷</td>
+              <td className="number">
+                <button type="button" onClick={this.handleClick}>AC</button>
+              </td>
+              <td className="number">
+                <button type="button" operation="±" onClick={this.handleClick}>±</button>
+              </td>
+              <td className="number">
+                <button type="button" operation="%" onClick={this.handleClick}>%</button>
+              </td>
+              <td className="calcSign">
+                <button type="button" operation="÷" onClick={this.handleClick}>÷</button>
+              </td>
             </tr>
             <tr>
-              <td className="number">7</td>
-              <td className="number">8</td>
-              <td className="number">9</td>
-              <td className="calcSign">×</td>
+              <td className="number">
+                <button type="button" name="7" onClick={this.handleClick}>7</button>
+              </td>
+              <td className="number">
+                <button type="button" name="8" onClick={this.handleClick}>8</button>
+              </td>
+              <td className="number">
+                <button type="button" name="9" onClick={this.handleClick}>9</button>
+              </td>
+              <td className="calcSign">
+                <button type="button" operation="×" onClick={this.handleClick}>×</button>
```

Comment:
- [ ] I am unable to perform the multiply operation on your Calculator. Every time I do that your app crashes and disappears from the screen with console errors (see the first image). This is because (according to the second image), there seems to be a size issue with you `'x'`. In the second image, for each pair of 'x', the second 'x' (the correct one) is the one I copied from **`operate.js`** and you can see that it is different from your original 'x'. Kindly fix that.

  ![benjp-times-dis](https://user-images.githubusercontent.com/36542986/148346804-7a7622c6-435e-4b12-b8df-3399160e101d.jpg)
  ![benjp-times](https://user-images.githubusercontent.com/36542986/148346812-822bcbdd-3313-4522-8ff5-1877b232aacf.jpg)